### ISSUE TITLE
QAT - Quantization Aware Training

### ISSUE THREAD

**Issue by bratao** [axolotl-ai-cloud/axolotl#2540]
### ⚠️ Please check that this feature request hasn't been suggested before.

- [x] I searched previous [Ideas in Discussions](https://github.com/axolotl-ai-cloud/axolotl/discussions/categories/ideas) didn't find any similar feature requests.
- [x] I searched previous [Issues](https://github.com/axolotl-ai-cloud/axolotl/labels/enhancement) didn't find any similar feature requests.

### 🔖 Feature description

The Quantization Aware Training showed extraordinary performance for the latest Gemma 3. It would be nice if Axolotl supports it.

### ✔️ Solution

The torchtune support it using [torchao](https://github.com/pytorch/ao/blob/v0.2.0/torchao/quantization/prototype/qat.py) https://pytorch.org/torchtune/main/tutorials/qat_finetune.html

### Acknowledgements

- [x] My issue title is concise, descriptive, and in title casing.
- [x] I have searched the existing issues to make sure this feature has not been requested yet.
- [x] I have provided enough information for the maintainers to understand and evaluate this request.

**Comment by ehartford** [comment]
thank you :-)

**Comment by winglian** [comment]
#2590 is in progress and should land soon
